packages/components/pie-tag/src/index.ts

```diff
@@ -45,10 +45,50 @@ export class PieTag extends LitElement implements TagProps {
     @validPropertyValues(componentSelector, iconPlacements, defaultProps.iconPlacement)
     public iconPlacement = defaultProps.iconPlacement;
 
+    private isIconOnly = false;
+
+    updated (changedProperties: PropertyValues<this>) {
+        if (changedProperties.has('size')) this.checkIfIsIconOnly();
+    }
+
+    private checkIfIsIconOnly () {
+        // The size must be large
+        const isLargeSize = this.size === 'large';
+
+        // The default slot must be empty
+        const defaultSlotText = this.textContent?.trim();
+        const isDefaultSlotEmpty = defaultSlotText === '';
+
+        // The icon slot must have some content
+        const iconSlot = this.shadowRoot?.querySelector('slot[name="icon"]') as HTMLSlotElement | null;
```

Comment:
Do you think we could be more declarative and use Lit's `queryAssignedNodes` decorator here?

We usually use these decorators to query the slotted children in the codebase.

https://lit.dev/docs/components/shadow-dom/#query-assigned-nodes